chore(dw): add content to data warehouse scene

naumaanh opened the pull request:
## Problem

Incorporated the changes from -> https://github.com/PostHog/posthog/pull/36474 

This adds the scorecards, the mat. view table, the recent activity table, the data sources table, and a few other misc buttons

## Changes
Frontend
- used lemontables , tooltips to build out main scene view
- used kea loaders for activity and total row stats
- adjusted for better type safety
- added auto-load on last page for recent activity

Backend
- added backend endpoints for total row stats, recent activity w/ pagination support
- updated backend response structures
- added tests 

<img width="1388" height="999" alt="Screenshot 2025-08-20 at 7 45 59 PM" src="https://github.com/user-attachments/assets/467b980f-17a7-4115-b85e-7387e0697222" />

## How did you test this code?

Manually on the frontend and several tests

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

naumaanh committed:
cleanup

naumaanh committed:
centralize .job() call

naumaanh committed:
Merge branch 'master' into naumaan/dwh-scene

naumaanh committed:
adjust ci typecheck error

naumaanh committed:
remove refresh button

naumaanh committed:
more cleanup, more ci issues

naumaanh committed:
more adjustments

naumaanh committed:
add row counts

naumaanh committed:
adjust for type safety and remove as any

naumaanh committed:
change to the existing posthog pagianation system via lemon ui

naumaanh committed:
removing dead code (search functionality)

naumaanh committed:
removing hardcoding for source types

naumaanh committed:
update wording

naumaanh committed:
Merge branch 'master' into naumaan/dwh-scene

naumaanh committed:
correct rows synced

naumaanh committed:
design and api update

naumaanh committed:
removing code smells

naumaanh committed:
fix

naumaanh committed:
fix

naumaanh committed:
adding greptile suggestions

naumaanh committed:
adding greptile suggestion

naumaanh committed:
remove billing attempt

naumaanh committed:
updated logic for total rows processed

naumaanh committed:
simplify logic and add a tooltip

naumaanh committed:
Merge branch 'master' into naumaan/dwh-scene

github-actions[bot] committed:
Update UI snapshots for `chromium` (4)

github-actions[bot] committed:
Update UI snapshots for `chromium` (10)

github-actions[bot] committed:
Update UI snapshots for `chromium` (4)

github-actions[bot] committed:
Update UI snapshots for `chromium` (10)

naumaanh committed:
fix kea logic, add data normalization for sourcetype, and adjust frontend to use lemontables instead of cards

naumaanh committed:
move logic to kea and switch to MTD instead of lifetime

naumaanh committed:
simplify mtd

naumaanh committed:
cleanup table column types and add tooltips

naumaanh committed:
inital attempt for moving rows calculaiton to backend

naumaanh committed:
swap to validationerror instead of generic raise

naumaanh committed:
adding fallback

naumaanh committed:
Merge branch 'master' into naumaan/dwh-scene-backend-stats-for-total-rows

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

naumaanh committed:
Merge branch 'master' into naumaan/dwh-scene-backend-stats-for-total-rows

naumaanh committed:
refine the structure of the results to seperate mat. view rows from pipeline source rows

naumaanh committed:
Merge branch 'master' into naumaan/dwh-scene-backend-stats-for-total-rows

naumaanh committed:
ci fix

naumaanh committed:
Merge branch 'naumaan/dwh-scene-backend-stats-for-total-rows' of https://github.com/PostHog/posthog into naumaan/dwh-scene-backend-stats-for-total-rows

naumaanh committed:
Merge branch 'master' into naumaan/dwh-scene-backend-stats-for-total-rows

naumaanh committed:
move filters into the try block

naumaanh committed:
Merge branch 'naumaan/dwh-scene-backend-stats-for-total-rows' of https://github.com/PostHog/posthog into naumaan/dwh-scene-backend-stats-for-total-rows

naumaanh committed:
actual fix this time lol

naumaanh committed:
add Eric's suggestions for using billing interval object, and defaulting to 0 if billing object doesn't exist

naumaanh committed:
Adding Eric's suggestion for billing tracked data and aggregated amount that hasn't been included in billing sync yet

naumaanh committed:
make billing_interval initialy an empty string. no need to populate it

naumaanh committed:
set fallback to match expected style, month vs monthly

naumaanh committed:
adding Tom's suggestions

naumaanh committed:
Merge branch 'master' into naumaan/dwh-scene-backend-stats-for-total-rows

naumaanh committed:
fix security issue

naumaanh committed:
add a test and some refinements

naumaanh committed:
cleanup

naumaanh committed:
change else to show defaults, and add a warning

naumaanh committed:
undo recent change since it wasn't actually needed

naumaanh committed:
Merge branch 'master' into naumaan/dwh-scene-backend-stats-for-total-rows

github-actions[bot] committed:
Update UI snapshots for `chromium` (8)

github-actions[bot] committed:
Update UI snapshots for `chromium` (8)

naumaanh committed:
fix tests

naumaanh committed:
Merge branch 'naumaan/dwh-scene-backend-stats-for-total-rows' of https://github.com/PostHog/posthog into naumaan/dwh-scene-backend-stats-for-total-rows

naumaanh committed:
adjust test import to fix python code quality check

naumaanh committed:
Merge branch 'master' into naumaan/dwh-scene-backend-stats-for-total-rows

naumaanh committed:
fix typo, add billing available object, and adjust tests for typo

naumaanh committed:
Merge branch 'naumaan/dwh-scene-backend-stats-for-total-rows' of https://github.com/PostHog/posthog into naumaan/dwh-scene-backend-stats-for-total-rows

naumaanh committed:
Merge branch 'master' into naumaan/dwh-scene-backend-stats-for-total-rows

naumaanh committed:
adding backend api progress

naumaanh committed:
check failures

naumaanh committed:
data adjustments and tests

naumaanh committed:
forgot the tests lol

naumaanh committed:
lean out the implemntation

naumaanh committed:
implement pagination for api

naumaanh committed:
fix ci issues

naumaanh committed:
Merge branch 'master' into naumaan/dwh-scene-backend-recent-activity

naumaanh committed:
fix merge

naumaanh committed:
semantic update

naumaanh committed:
fix casing, adjust tests

naumaanh committed:
update test

naumaanh committed:
refine activities

naumaanh committed:
Merge branch 'master' into naumaan/dwh-scene-backend-recent-activity

naumaanh committed:
clean up pagination

naumaanh committed:
Merge branch 'naumaan/dwh-scene-backend-recent-activity' of https://github.com/PostHog/posthog into naumaan/dwh-scene-backend-recent-activity

naumaanh committed:
fix tests

naumaanh committed:
add offset back

github-actions[bot] committed:
Update UI snapshots for `chromium` (15)

github-actions[bot] committed:
Update UI snapshots for `chromium` (15)

naumaanh committed:
correct total rows test to look for correct snake case fields

naumaanh committed:
Merge branch 'naumaan/dwh-scene-backend-recent-activity' of https://github.com/PostHog/posthog into naumaan/dwh-scene-backend-recent-activity

naumaanh committed:
greptile suggestion

naumaanh committed:
remove greptile suggestion

naumaanh committed:
setting a 250 max

naumaanh committed:
Merge branch 'master' into naumaan/dwh-scene

naumaanh committed:
bring in changes from activity branch

naumaanh committed:
adjust to use backend calculations

naumaanh committed:
Merge branch 'master' into naumaan/dwh-scene-backend-recent-activity

naumaanh committed:
cleanup error messaging

naumaanh committed:
Merge branch 'master' of https://github.com/PostHog/posthog into naumaan/dwh-scene-backend-recent-activity

naumaanh committed:
change ot using union query

naumaanh committed:
add updated tests

naumaanh committed:
fix list type

naumaanh committed:
Merge branch 'master' into naumaan/dwh-scene-backend-recent-activity

naumaanh committed:
fix source_type in sql expression

naumaanh committed:
Merge branch 'master' into naumaan/dwh-scene-backend-recent-activity

naumaanh committed:
change source form Data sync to source value

naumaanh committed:
change source form Data sync to source value

naumaanh committed:
fix tests to match expected data

naumaanh committed:
Merge branch 'master' into naumaan/dwh-scene-backend-recent-activity

naumaanh committed:
Merge branch 'master' into naumaan/dwh-scene

naumaanh committed:
Merge branch 'naumaan/dwh-scene-backend-recent-activity' into naumaan/dwh-scene

naumaanh committed:
adding pagination response method to api, adjusting types, and adding more tests

naumaanh committed:
cleanup

naumaanh committed:
Merge branch 'naumaan/dwh-scene-backend-recent-activity' into naumaan/dwh-scene

naumaanh committed:
update pagination contorls to fetch more data for recent activity when at the last page

naumaanh committed:
Merge branch 'master' into naumaan/dwh-scene-backend-recent-activity

naumaanh committed:
cleanup

naumaanh committed:
Merge branch 'naumaan/dwh-scene-backend-recent-activity' into naumaan/dwh-scene

naumaanh committed:
linking data from backend, and cleanup

naumaanh committed:
Merge branch 'master' into naumaan/dwh-scene

naumaanh committed:
fix tests and remove autocompletes

naumaanh committed:
Merge branch 'naumaan/dwh-scene' of https://github.com/PostHog/posthog into naumaan/dwh-scene

naumaanh committed:
Merge branch 'master' into naumaan/dwh-scene

naumaanh committed:
update look and feel for quick actions

naumaanh committed:
update look and feel for quick actions

naumaanh committed:
add greptile suggestions

naumaanh committed:
Merge branch 'naumaan/dwh-scene' into naumaan/dwh-quick-actions

naumaanh committed:
Merge branch 'master' into naumaan/dwh-scene

naumaanh committed:
Merge branch 'master' into naumaan/dwh-quick-actions

naumaanh committed:
Merge branch 'master' into naumaan/dwh-scene

naumaanh committed:
Merge branch 'master' into naumaan/dwh-quick-actions

naumaanh committed:
cleanup

github-actions[bot] committed:
Update UI snapshots for `chromium` (13)

naumaanh committed:
cleanup?

naumaanh committed:
cleanup?

naumaanh commented:
> will review after rebase so the changes from other PR are reflected here

Could you review this one first when you get a chance? -> https://github.com/PostHog/posthog/pull/36474 (this one has all the other UI element besides the quick action. the main part here is the actual scene UI logic, and then hooking up the backend/kea to work for paginating the recent activities. 

in comparison, this one builds off of that!

naumaanh committed:
Merge branch 'naumaan/dwh-quick-actions' of https://github.com/PostHog/posthog into naumaan/dwh-quick-actions

naumaanh committed:
Merge branch 'master' into naumaan/dwh-quick-actions

naumaanh committed:
Merge branch 'master' into naumaan/dwh-quick-actions

EDsCODE commented on frontend/src/scenes/data-warehouse/externalDataSourcesLogic.ts:
```diff
@@ -53,13 +59,102 @@ export const externalDataSourcesLogic = kea<externalDataSourcesLogicType>([
+                loadRecentActivity: () => 1,
+            },
+        ],
+        recentActivity: [
```
Can make this a [loader](https://keajs.org/docs/plugins/loaders/)

naumaanh replied:
moved it to a loader as `recentActivityResponse`!

EDsCODE commented on frontend/src/scenes/data-warehouse/DataWarehouseScene.tsx:
```diff
@@ -1,44 +1,344 @@
+    const { materializedViews } = useValues(dataWarehouseSceneLogic)
+    const { activityPaginationState } = useValues(externalDataSourcesLogic)
+    const { setActivityCurrentPage } = useActions(externalDataSourcesLogic)
+    const { computedAllSources, totalRowsStats } = useValues(dataWarehouseSettingsLogic)
```
is there a reason we are putting all the stats in various places. I think it might make sense to put all the api calls in datawarehousescenelogic. I dont think these stats and such will be used in other pages so it wouldn't make sense to have them loaded when, for example, dataWarehouseSettingsLogic is loaded somewhere else

naumaanh replied:
i had originally split it out thinking it might be reused in a couple of different places, but it makes more sense atm to collapse it all down

EDsCODE reviewed (commented):
Can we remove the quick actions and get the data warehouse scene updates in first (using the APIs you've created). The quick actions don't really make sense to me yet because right now they would just be encouraging you to sync more from sources you've already set up which isn't super necessary (i would guess most folks won't have > 1 stripe source)

naumaanh committed:
remove quick actions

naumaanh committed:
centralizing stats

naumaanh committed:
Merge branch 'master' into naumaan/dwh-quick-actions

naumaanh committed:
refactor: rename and restructure recent activity loading logic

- Updated action names from `loadRecentActivitySuccess` to `loadRecentActivityResponseSuccess` for clarity.
- Changed loader state management to accommodate new response structure.
- Introduced selectors for `recentActivity` and `recentActivityHasMore` based on the new response format.

naumaanh committed:
Merge branch 'naumaan/dwh-quick-actions' of https://github.com/PostHog/posthog into naumaan/dwh-quick-actions

github-actions[bot] committed:
Update UI snapshots for `chromium` (4)

naumaanh committed:
Merge branch 'master' into naumaan/dwh-quick-actions

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

naumaanh committed:
Merge branch 'master' into naumaan/dwh-quick-actions

naumaanh committed:
Merge branch 'master' into naumaan/dwh-quick-actions

EDsCODE commented on frontend/src/scenes/data-warehouse/settings/dataWarehouseSettingsLogic.ts:
```diff
@@ -23,6 +24,8 @@ export const dataWarehouseSettingsLogic = kea<dataWarehouseSettingsLogicType>([
             externalDataSourcesLogic,
             ['dataWarehouseSources', 'dataWarehouseSourcesLoading'],
+            availableSourcesDataLogic,
+            ['availableSources'],
```
unneeded change?

naumaanh replied:
thanks, you're correct here, this isn't needed!

EDsCODE commented on frontend/src/scenes/data-warehouse/externalDataSourcesLogic.ts:
```diff
@@ -62,4 +62,7 @@ export const externalDataSourcesLogic = kea<externalDataSourcesLogicType>([
             }
         },
     })),
+    afterMount(({ actions }) => {
```
If we're loading it specifically on datawarehousescene logic, I think this change won't be needed

EDsCODE commented on frontend/src/scenes/data-warehouse/DataWarehouseScene.tsx:
```diff
@@ -1,44 +1,291 @@
+
+const LIST_SIZE = 5
 
 export function DataWarehouseScene(): JSX.Element {
```
Need some loading states. The tables are just "no entries" while it's loading. Doesn't have to be in this PR but should follow up with it

naumaanh replied:
added a global loading state variable and plugged that into the lemon table preset `loading` connector and added `loadingSkeletonRows`

EDsCODE commented on frontend/src/scenes/data-warehouse/DataWarehouseScene.tsx:
```diff
@@ -1,44 +1,291 @@
+
+import { dataWarehouseSceneLogic } from './dataWarehouseSceneLogic'
+
+export const scene: SceneExport = { component: DataWarehouseScene }
```
you can add the dataWarehouseSceneLogic here so that it mounts alongside the scene

EDsCODE commented on frontend/src/scenes/data-warehouse/DataWarehouseScene.tsx:
```diff
@@ -1,44 +1,291 @@
+    const sourcesPagination = usePagination(computedAllSources, { pageSize: LIST_SIZE }, 'sources')
+    const viewsPagination = usePagination(materializedViews || [], { pageSize: LIST_SIZE }, 'views')
+
+    const activityPagination = {
```
~unfamiliar with the paginating controls. Why is this pagination handled differently than the above usePagination options~ nvm I see. We'll probably want something like this for the above too but ok omitting for now

naumaanh replied:
sounds good, happy to change convention here as well. 

I know the action in the syncs page for the source is a load more button instead of paginating through the responses, so if thats a more preferred mechanism, can switch to that!

EDsCODE reviewed (commented):
A bit more cleanup

Note: Another improvement that needs to happen that's not directly required in this PR but when you navigate to the "create new source" form and hit cancel, it doesn't return you to this page. This is a more general issue that needs to be solved as we trigger the form in various places and don't respect the return

naumaanh committed:
feat(data-warehouse): add loading state for tables and views in DataWarehouseScene

- Integrated loading state into DataWarehouseScene using tablesLoading.
- Updated dataWarehouseSceneLogic to include tablesLoading derived from databaseLoading and dataWarehouseSourcesLoading.
- Removed unnecessary afterMount action in externalDataSourcesLogic.
- Cleaned up imports in dataWarehouseSettingsLogic by removing unused availableSourcesDataLogic.

naumaanh committed:
Merge branch 'master' into naumaan/dwh-quick-actions

naumaanh committed:
Merge branch 'master' into naumaan/dwh-quick-actions

naumaanh committed:
Merge branch 'master' into naumaan/dwh-quick-actions

naumaanh committed:
Merge branch 'master' into naumaan/dwh-quick-actions
